### PR TITLE
Ai/day5

### PR DESCRIPTION
**Pull request by agherasie**
# Description

Please provide a detailed description of what was done in this PR.
Precise the issue that you are resolving.

# Changes include

- [ ] Bugfix (non-breaking change that solves an issue)
- [ ] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (change that is not backwards-compatible and/or changes current functionality)

# Breaking changes

Please complete this section if any breaking changes have been made, otherwise delete it.

# Checklist

- [ ] I have assigned this PR to myself
- [ ] I have added at least 1 reviewer
- [ ] I have added the needed labels
- [ ] I have linked this PR to an issue
- [ ] I have linked this PR to a milestone
- [ ] I have linked this PR to a project
- [ ] I have tested this code
- [ ] I have added / updated tests (unit / functionals / end-to-end / ...)
- [ ] I have updated the README and other relevant documents (guides...)
- [ ] I have added sufficient documentation both in code, as well as in the READMEs

# Additional comments

Please post additional comments in this section if you have them, otherwise delete it.

### CODE REVIEW

File: Artificial Intelligence/day05/1.Introduction/Q_Learning.ipynb [hunk 1]
```diff
@@ -0,0 +1,529 @@
+{
+ "cells": [
+  {
+   "attachments": {},
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Value-based method: Q Learning\n",
+    "\n",
+    "In the first notebook of the day, we will learn about one of the most popular RL frameworks: Q Learning !\n",
```
Comment (Mikatech) [inline Artificial Intelligence/day05/1.Introduction/Q_Learning.ipynb:10]:
```suggestion
    "In the first notebook of the day, we will learn about one of the most popular RL algorithm: Q Learning !\n",
```

**Review by Mikatech — COMMENTED** [review]
Add requirements.txt

File: Artificial Intelligence/day05/1.Introduction/Q_Learning.ipynb [hunk 2]
```diff
@@ -0,0 +1,529 @@
...
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "class Agent:\n",
+    "    \"\"\"\n",
+    "    This class defines our Agent which will interact with the environment and update its Q Table\n",
+    "    \"\"\"\n",
+    "    \n",
+    "    def __init__(self):\n",
+    "        # We initialize a value called `epsilon` (we will soon learn more about it)\n",
```
Comment (Mikatech) [inline Artificial Intelligence/day05/1.Introduction/Q_Learning.ipynb:152]: You did not talk about epsilon, so what i initialize ? I let none ?

**Comment by Mikatech** [comment]
I think you should add ``assert`` to allow students to verify that their solutions works

File: Artificial Intelligence/day05/1.Introduction/Q_Learning.ipynb [hunk 3]
```diff
@@ -0,0 +1,529 @@
...
+  {
+   "attachments": {},
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Now, let's create an `Agent` class which will contain our action values and a method to update them !\n",
+    "\n",
+    "In Q-Learning, the table which contains our action values is called the Q-Table ! (catchy, right ?)\n",
+    "\n",
+    "In order to define this Q-Table, we simply create an array of size `number_of_states * number_of_actions` and initialize all of its values to 0.\n",
```
Comment (Mikatech) [inline Artificial Intelligence/day05/1.Introduction/Q_Learning.ipynb:125]:
```suggestion
    "In order to define this Q-Table, we simply create an array of shape `(number_of_states, number_of_actions)` and initialize all of its values to 0.\n",
```

**Comment by agherasie** [comment]
> I think you should add `assert` to allow students to verify that their solutions works

**Pull request merged**